Reinstate G Suite deployment docs (#5929)
sourcegraph/sourcegraph: doc/integration/google_gsuite.md

# G Suite and Chrome Enterprise integration

> NOTE: Company-wide deployment via G Suite is a [paid upgrade](https://about.sourcegraph.com/pricing)

You can install and preconfigure the Sourcegraph Chrome extension for all users in your G Suite organization, so that they can get code intelligence on files and diffs.

## Automatically install with G Suite

> Using the Google Admin Console, you can [force install](https://support.google.com/chrome/a/answer/6306504?hl=en) the Sourcegraph browser extension for all users in your organization. To do so:

Reviewer: ```suggestion
Using the Google Admin Console, you can [automatically install](https://support.google.com/chrome/a/answer/6306504?hl=en) the Sourcegraph browser extension for all users in your organization. To do so:
```

1. Visit the Google Admin Console [Chrome management](https://admin.google.com/u/2/ac/chrome/apps/user) page, and click "Apps and extensions".
2. Select the name of your organization in the left-hand menu.
3. Click the **+** button to add an extension, select "Add Chrome app or extension by ID", and paste in the ID of the Sourcegraph browser extension: `dgjhfomjieaadpoljlnidmbgkdffpack`.
4. In "Installation Policy", select "Force install".
5. In the "Policy for extensions" field, fill in a JSON value containing the URL of your self-hosted Sourcegraph instance:

```
{
    "sourcegraphURL": {
        "Value": "https://sourcegraph.example.com"
    }
}
```
6. Click "Save".

The Sourcegraph browser extension will then be rolled out to all managed browsers in your organization, preconfigured to point to your self-hosted Sourcegraph instance. The roll-out may take some time.

> Users will need to be logged in to your self-hosted Sourcegraph instance for the browser extension to provide code intelligence.

Reviewer: ```suggestion
Users will need to have valid accounts and be signed in to your Sourcegraph instance for the browser extension to provide code intelligence.
```

# G Suite and Chrome Enterprise integration

> NOTE: Company-wide deployment via G Suite is a [paid upgrade](https://about.sourcegraph.com/pricing)

You can install and preconfigure the Sourcegraph Chrome extension for all members of a G Suite organization or organizational unit, so that they can get code intelligence on files and diffs.

> Users can disable Sourcegraph at any time by using the toggle in the extension settings:

Reviewer: Sorry, but reading this now, this feels a bit out of place... Can we move this sentence and the screenshot to the very bottom instead?

Author: Done.

<img src="img/disable_extension.png" width="400">

> By default, the browser extension only has access to github.com. Access to additional sites is granted by the user, on a site-by-site basis.

Reviewer: Is this true even for G Suite deployments? That's actually sort of annoying haha. There's no way for an admin to configure this?

Reviewer: I.e., would each member of the team have to click on the icon to grant access to the code host, if it wasn't github.com?

Author: Yes, this is annoying. I found no way for admins of directly configuring permissions, and the display of the permissions popup has to be triggered by user actions (so just setting an array of `codeHostUrls` through managed storage and calling `permissions.request()` for these URLs wouldn't work).

We could work around this by hinting to the user that he should open the popup (eg. by badging the icon) to grant additional permissions if code host URLs have been set through managed storage.
